flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/declarative/allocator/SlotAllocator.java

```diff
@@ -30,12 +31,12 @@
      * Determines the parallelism at which the vertices could given the collection of slots.
      *
      * @param jobInformation information about the job graph
-     * @param freeSlots currently free slots
+     * @param slots Slots to consider for determining the parallelism
      * @return parallelism of each vertex along with implementation specific information, if the job
      *     could be run with the given slots
      */
     Optional<T> determineParallelism(
-            JobInformation jobInformation, Collection<SlotInfoWithUtilization> freeSlots);
```

Comment:
I want to call `determineParallelism` with ALL slots, instead of the free slots. I believe we should not distribute increments of new slots, but always the total number of available slots.
```java
slotAllocator.determineParallelism(
                                new JobGraphJobInformation(jobGraph),
                                declarativeSlotPool.getAllSlotsInformation());
```

Maybe we need to revisit the DeclarativeSlotPool.getAllSlotsInformation() method? I guess we could also calculate the taskExecutorUtilization for allocated slots as well?

Reply:
Yes, we could look into changing this method, but ideally we get by without having to modify the SlotPool interface in significant ways.
Alternatively we could always compute it within the scheduler on the fly. That may be a better option than forcing this computation upon all usages of this method.

Reply:
I'm not sure if computing it in the scheduler is the right separation of concerns. For calculating the utilization, I need to know whether the slot is free or allocated. That's something that the scheduler can not get from the `SlotInfo` interface.

What's the problem with modifying the SlotPool interface? (Is changing the return type of a method significant?)

Reply:
You would then need to touch the SlotpoolImpl which would be a waste of time, both for developers and in terms of performance (unless you go with a broken implementation that always returns 0, but, you know...).

Computing it one the fly isn't that difficult; you get the free slots, extract utilization by task manager, and then query it for all slots returned by getAllSlotsInformation.

Nevertheless, I think the best short-term solution is to stick with plain `SlotInfos`. There's no immediate need for the utilization, and whether we solve it now or later probably won't change anything.

Reply:
Allright, then let's change this once we need the utilization in the new scheduler.